.github/workflows/try.yml

```diff
@@ -20,13 +24,15 @@ jobs:
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
-      #- uses: actions/checkout@master
-      #  with:
-      #    name: qiboteam/qibo 
-      - run: git pull
-      - run: git clone https://github.com/qiboteam/qibo.git  
-      - run: ls -l
+
+      - name: Clone qibo repository
+        uses: actions/checkout@v2
+        with:
+          repository: qiboteam/qibo
+          path: "~/"
+      - run: git clone https://github.com/qiboteam/qibo.git
       - name: Commit & Push changes
-        uses: actions-js/push@master
+        uses: actions-js/push@work2
```

Comment:
Avoid as much as possible little used actions (the might be discontinued at any time). This in particular has only 39 stars.

It seems "almost official", because the organization is `actions-js`, but in fact it is a random one (`actions` is the official one).

Reply:
Moreover: do not push on the repository, especially not on the official branch.

One option would be to actually push the generated stuff to a separate branch, and deploy from there. Until a few months ago was the only option for GitHub Pages, but now there is a better one https://github.com/actions/upload-pages-artifact

It is better because the repository should contain only manually written content, while the generated stuffs should be kept separate, and just make it simple to regenerate.
Essentially, they should follow the version of the generator, and not be versioned separately. The best way to make sure there is no such a discrepancy, is not to version generated objects at all, and just version the generator.
This will keep also repos smaller (and it is a good thing, since they tend to grow in the long run, especially including generated objects, that are much bigger than what you write manually).